org.adempiere.base.process/src/org/compiere/process/M_PriceList_Create.java

```diff
@@ -295,7 +299,9 @@ else if (M_Pricelist_Version_Base_ID == 0) {
 						+ " INNER JOIN M_DiscountSchemaLine dl ON (dl.M_DiscountSchemaLine_ID=?) "	//	#3
 						//
 						+ "WHERE EXISTS (SELECT * FROM T_Selection s WHERE s.AD_PInstance_ID=? AND po.M_Product_ID=s.T_Selection_ID)"	//	#4
-						+ " AND po.IsCurrentVendor='Y' AND po.IsActive='Y'";
+						+ " AND ((dl.C_BPartner_ID IS NULL AND po.IsCurrentVendor='Y')"
+						+ "      OR (po.C_BPartner_ID=dl.C_BPartner_ID AND (dl.IsIgnoreIsCurrentVendor='Y' OR po.IsCurrentVendor='Y')))"
+						+ " AND po.IsActive='Y'";

```

Comment:
for consistency, should have added another "AND " clause here for VendorCategory

Reply:
but it's not necessary, because there may be only one po-line for a given bpartner or one with IsCurrentVendor so because T_Selection contains only lines where VendorCategory matches it will match here anyway.